### DISCUSSION TITLE
Upper bound on YALMIP and time constraints

### DISCUSSION THREAD

**Discussion by ArjunM21** [yalmip/YALMIP#1187] [Optimization modelling]
1. When using YALMIP, sedumi starts the branch and bound algorithm from infinity, I was wondering if it's possible to give an upper bound to the function at hand, which can reduce the number of computations required. What I mean is that sedumi starts the algorithm from the upper bound and not infinity. 

2. In YALMIP, if the solver runs out of 3600 seconds, it is terminated and the value at that moment becomes the solution. I was wondering if it's possible to change either the total time taken by the solver or the tolerance value chosen to check for optimality of the solver.

**Comment by johanlofberg** [comment]
sedumi is not a branch&bound solver, so I think there is some confusion here. 

Perhaps you re talking about bnb. To find out available options, simply display the bnb options structure, or use help bnb

**Reply by ArjunM21** [reply]
Hey I'm sorry for my error. I meant bnb in yalmip. I did try to get all the options, but wasn't able to see any options with regards to upper bound or such. 

What I require is to give an upper bound to bnb so that the solver starts solving from the upper bound instead of infinity as always. Do you think that is possible in some other way, if not directly?

**Reply by johanlofberg** [reply]
You use assign and usex0 as always
https://yalmip.github.io/warmstart

**Reply by ArjunM21** [reply]
Thank you for an answer. But unfortunately, this is not working in bnb solver for some reason. Pls see part of the code I used it in. Pls also note that I have LMI constraints in my code.

`
q = binvar(nv,1);
Q = diag(q);
z = binvar(nv,1);
assign(q, round(z));
ops = sdpsettings('usex0',1);
`

Is there a reason why this is not working? Have I missed something? I went through the document you shared, but couldn't figure anything else out.

**Reply by johanlofberg** [reply]
Assigning a function of a decision variable as an initial guess makes no sense. An initial guess is a constant value.

**Reply by ArjunM21** [reply]
Okay thank you. Here, the command of assign is done after the z values are determined. So if I give a value to q, which would be a nvx1 vector, this should work?

**Reply by johanlofberg** [reply]
Reasonable code would be

```
q = binvar(nv,1); Q = diag(q); 
my_guess = [1;0;1;0;...]
assign(q, my_guess); 
ops = sdpsettings('usex0',1);
optimize(constraints,obj,ops)
```

**Reply by ArjunM21** [reply]
Yeah I understood that. But it did not work even when I tried this. Actually the solver still starts from inf and not the initial guess that I have provided. Will this work for bnb?

**Reply by johanlofberg** [reply]
You would have to supply a reproducible example. If you have assigned an initial solution and turned on warm-start, the initial upper bound should not be inf
